sycl/plugins/level_zero/pi_level_zero.cpp

```diff
@@ -625,28 +652,81 @@ ze_result_t ZeCall::doCall(ze_result_t ZeResult, const char *ZeName,
   if (!(condition))                                                            \
     return error;
 
+// Returns bool value indicating whether queue supports in-order optimization
+// for provided type of event.
+bool _pi_queue::supportsInOrderQueueOptimization(bool HostVisible,
+                                                 bool IsDiscarded) {
+  if (DisableEventsCaching)
+    return false;
+
+  bool ProfilingEnabled = (Properties & PI_QUEUE_PROFILING_ENABLE) != 0;
```

Comment:
Why can't we reuse events that have no external references to even if queue is with profiling?

Reply:
I guess we can reuse, I will fix.

Reply:
Fixed.